app/views/declaration/previousDocuments/previous_documents_edit.scala.html

```diff
@@ -0,0 +1,128 @@
+        saveButtons: saveButtons,
+        formHelper: uk.gov.hmrc.play.views.html.helpers.FormWithCSRF
+)
+
+@(mode: Mode, form: Form[Document])(implicit request: JourneyRequest[_], messages: Messages)
+
+@tariffBody = { @Html(messages("declaration.type.previousDocumentsSummaryText", components.details_content_link())) }
+
+@pageHint = {
+    <span class="govuk-hint">
+        @messages("declaration.previousDocuments.hint")
+    </span>
+}
+
+@documentCategory = {
+    @govukRadios(Radios(
+        name = "documentCategory",
+        fieldset = Some(Fieldset(
+            legend = Some(Legend(
+                content = Text(messages("declaration.previousDocuments.reasonQuestion")),
+                isPageHeading = false,
+                classes = "govuk-label--m"
+            ))
+        )),
+        items = Seq(
+            RadioItem(
+                id = Some("simplified-declaration"),
+                value = Some(SimplifiedDeclaration.value),
+                content = Text(messages("declaration.previousDocuments.Y")),
+                checked = form("documentCategory").value.contains(SimplifiedDeclaration.value)
+            ),
+            RadioItem(
+                id = Some("related-document"),
+                value = Some(RelatedDocument.value),
+                content = Text(messages("declaration.previousDocuments.Z")),
+                checked = form("documentCategory").value.contains(RelatedDocument.value)
+            )
+        ),
+        errorMessage = form("documentCategory").error.map(err => ErrorMessage(content = Text(messages(err.message, err.args:_*))))
+    ))
+}
+
+@documentType = {
+    @components.fields.field_accessible_autocomplete(
+        field = form("documentType"),
+        label = "declaration.previousDocuments.documentType",
+        labelClass = Some("govuk-label--m"),
+        hintText = None,
+        emptySelectValue = messages("declaration.previousDocuments.documentType.empty"),
+        items = AutoCompleteItem.fromDocumentType(DocumentType.allDocuments)
+    )
+}
+
+@documentReference = {
+    @exportsInputText(
+        field = form("documentReference"),
+        labelKey = "declaration.previousDocuments.documentReference",
+        hintKey = Some("declaration.previousDocuments.documentReference.hint")
+    )
+}
+
+@goodsItemIdentifier = {
+    @exportsInputText(
+        field = form("goodsItemIdentifier"),
+        labelKey = "declaration.previousDocuments.goodsItemIdentifier",
+        hintKey = Some("declaration.previousDocuments.goodsItemIdentifier.hint"),
+        inputClasses = Some("govuk-input--width-2")
+    )
+}
+
+
+@errorSummary(form.errors)
+
+@sectionHeader(messages("declaration.previousDocuments.heading"))
+
+@govukFieldset(Fieldset(
+    legend = Some(Legend(
+        content = Text(messages("declaration.previousDocuments.title")),
+        classes = gdsPageLegend,
+        isPageHeading = true
+    )),
+    html = HtmlFormat.fill(immutable.Seq(
```

Comment:
Nit: I don't think you need `immutable` prefix. Immutable implementations are preferable and default. It is a bit misleading (at least to me) that you have this prefix. If you write with prefix I'd expect it to be `mutable`.

Reply:
We have that pattern all over the place where HtmlFormat.fill is used.  Not sure why though.  I'll have a look.